From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct fib_dump_filter
Patch-mainline: never, kabi
References: kabi

In networking-stable-20_05_27, commit 41b4bd986f86 (net: don't return
invalid table id error when we fall back to PF_UNSPEC) removed a member
from struct fib_dump_filter. This indeed changed the layour of the
structure and the kABI checker complains now.

struct fib_dump_filter can be used in other structure, so add back this
internal-only member and be done with that.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/ip_fib.h |    1 +
 1 file changed, 1 insertion(+)

--- a/include/net/ip_fib.h
+++ b/include/net/ip_fib.h
@@ -245,6 +245,7 @@ struct fib_dump_filter {
 	u32			table_id;
 	/* filter_set is an optimization that an entry is set */
 	bool			filter_set;
+	bool                    dump_all_families; // kABI
 	bool			dump_routes;
 	bool			dump_exceptions;
 	unsigned char		protocol;
